### ISSUE TITLE
Can't match on sigmaEq2 Refl Refl: regression ?

### ISSUE THREAD

**Issue by nicolabotta** [idris-lang/Idris-dev#2261]
The code
```
> %default total

> ||| Introduction
> sigmaEq2 : {A : Type} -> 
>            {P : A -> Type} -> 
>            {s1: Sigma A P} -> 
>            {s2: Sigma A P} ->
>            getWitness s1 = getWitness s2 ->
>            getProof s1 = getProof s2 ->
>            s1 = s2
> sigmaEq2 {A} {P} {s1 = (a ** p)} {s2 = (a ** p)} Refl Refl = Refl
```
used to type check in 0.9.17.1-git:09a0a8d but with 0.9.17.1-git:d215440 one gets
```
- + Errors (1)
 `-- sigma_equality.lidr line 11 col 9:
     When elaborating left hand side of sigmaEq2:
     Can't match on sigmaEq2 Refl Refl
```

**Comment by nicolabotta** [comment]
David Feuer has suggested the following workaround:
```
> sigmaEq2 {s1=(a ** p)} {s2 = (a' ** p')} pf1 pf2 with (pf1,pf2)
>   sigmaEq2 {s1=(a ** p)} {s2 = (a ** p)} pf1 pf2 | (Refl, Refl) = Refl
```
Thanks, David! Still this seems a regression.

**Comment by nicolabotta** [comment]
Closed by Idris version 0.9.18-git:3034fb6
